external/kafka/src/main/scala/org/apache/gearpump/streaming/kafka/lib/consumer/KafkaMessage.scala

```diff
@@ -16,10 +16,17 @@
  * limitations under the License.
  */
 
-package org.apache.gearpump.streaming.kafka.lib
+package org.apache.gearpump.streaming.kafka.lib.consumer
 
 import kafka.common.TopicAndPartition
 
+/**
+ * wrapper over messages from kafka
+ * @param topicAndPartition where message comes from
+ * @param offset message offset on kafka queue
+ * @param key message key, could be None
+ * @param msg message payload
+ */
 case class KafkaMessage(topicAndPartition: TopicAndPartition, offset: Long,
```

Comment:
`KafkaMessage` is not transferred down streams but for `KafkaSource`. `topicAndPartition` would dispatch message to its corresponding `KafkaOffsetManager` and `offset` is stored along with the timestamp for replay.